frontend/src/plantlistpage/PlantListPage.test.js

```diff
@@ -0,0 +1,21 @@
+import { render } from '@testing-library/react'
+import 'jest-styled-components'
+import plants from '../data/plants.json'
+import PlantListPage from './PlantListPage'
+
+describe('PlantListPage', () => {
+  it('renders correctly', () => {
+    const { container } = render(<PlantListPage />)
+
+    expect(container.firstChild).toMatchSnapshot()
+  })
+
+  it('displays the plant names and species', () => {
+    const { getByText } = render(<PlantListPage />)
+
+    plants.forEach(({ name }) => expect(getByText(name)).toBeInTheDocument())
+    plants.forEach(({ species }) =>
+      expect(getByText(species)).toBeInTheDocument()
```

Comment:
Cooool! I haven't seen that one before!! I think I will copy it :)

Reply:
Copied it from the Student Companion App lel